front/app/containers/Admin/projects/project/general/index.tsx

```diff
@@ -376,68 +373,17 @@ const AdminProjectsProjectGeneral = () => {
               })
             : null;
 
-        const filesToAddPromises = projectFiles
-          .filter((file) => !file.remote)
-          .map((file) => {
-            if (latestProjectId) {
-              return addProjectFile({
-                projectId: latestProjectId,
-                file: {
-                  file: file.base64,
-                  name: file.name,
-                  ordering: file.ordering,
-                },
-              });
-            }
-
-            return;
-          });
-
-        const filesToRemovePromises = projectFilesToRemove
-          .filter((file) => file.remote === true && isString(file.id))
-          .map((file) => {
-            if (latestProjectId && file.id) {
-              return deleteProjectFile({
-                projectId: latestProjectId,
-                fileId: file.id,
-              });
-            }
-
-            return;
-          });
-
-        const reorderedFiles = projectFiles.filter((file) => {
-          const initialOrdering = file.id
-            ? initialProjectFilesOrdering[file.id]
-            : undefined;
-          return (
-            file.remote &&
-            typeof file.ordering !== 'undefined' &&
-            (typeof initialOrdering === 'undefined' ||
-              file.ordering !== initialOrdering)
-          );
-        });
-
-        const filesToReorderPromises = reorderedFiles.map((file) => {
-          if (latestProjectId && file.id) {
-            return updateProjectFile({
-              projectId: latestProjectId,
-              fileId: file.id,
-              file: {
-                ordering: file.ordering,
-              },
-            });
-          }
-          return;
+        await syncProjectFiles({
```

Comment:
Not sure how much it matters, but there  is a slight change in behavior here. Before, all the promises were fired at the same time and awaited together, so they could be handled in parallel. After this refactor, we are first updating the project files, waiting for that to be done, and only then updating the card image. This could potentially be a bit slower.

To restore the original behavior, you could do something like

```
const projectFilePromises = syncProjectFiles({
[...]

await Promise.all([
  [...]
  cardImageToRemovePromise,
  ...projectFilePromises
```